connector/spanmetricsconnector/config.go

```diff
@@ -71,7 +71,8 @@ type HistogramConfig struct {
 }
 
 type ExemplarsConfig struct {
-	Enabled bool `mapstructure:"enabled"`
+	Enabled             bool `mapstructure:"enabled"`
+	MaxNumPerDimensions int  `mapstructure:"max_num_per_dimensions"`
```

Comment:
I would rename this to `MaxPerDataPoint` which feels a bit more descriptive.

Also, please consider making the type a pointer to be more explicit about its optionality. I would prefer if it was some sort of "optional" type where we can call `IsSet()`, but maybe that's overkill.

Please also write a test for loading this config.